components/PodcastContainer.jsx

```diff
@@ -9,32 +10,59 @@ class PodcastContainer extends React.Component {
       title: PropTypes.string,
       link: PropTypes.string,
     })).isRequired,
+    lastPlayedEpisode: PropTypes.string,
   };
 
+  static defaultProps = {
+    lastPlayedEpisode: undefined,
+  }
+
   constructor(props) {
     super(props);
     this.state = {
-      currentEpisodeLink: this.getLatestEpisode().link,
+      currentEpisodeLink: this.getEpisodeToPlay().link,
     };
   }
 
-  getLatestEpisode() {
-    const { episodes } = this.props;
+  getEpisodeToPlay() {
+    // eslint-disable-next-line react/prop-types
+    const { episodes, lastPlayedEpisode } = this.props;
+    if (lastPlayedEpisode) {
+      const titles = episodes.map(episode => episode.title);
```

Comment:
Should we just compare the title to the `lastPlayedEpisode` here to find the index to avoid the extra array loop?

Reply:
Implemented using Array.some